src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -225,6 +231,50 @@ public void updateCheesesStatus(Set<CheeseId> cheesesAssigned) {
         cheeses.updateCheesesStatus(cheesesAssigned);
     }
 
+    /**
+     * Checks whether the address book's current state is valid
+     * Includes checking dependencies between models
+     */
+    public void checkAddressBook() {
+        Set<CustomerId> customerIdSet = customers.asUnmodifiableObservableList().stream()
+            .map(x -> x.getId()).collect(Collectors.toSet());
+        Map<CheeseId, Cheese> cheeseIdMap = cheeses.asUnmodifiableObservableList().stream()
+            .collect(Collectors.toMap(x -> x.getCheeseId(), x -> x));
+
+        HashSet<CheeseId> orderCheeseIdSet = new HashSet<>();
+        // Check that each order is valid on the system level
+        for (Order order : orders.asUnmodifiableObservableList()) {
+            CustomerId customerId = order.getCustomerId();
+            checkArgument(customerIdSet.contains(customerId), "Order "
+                + order.getOrderId() + "'s customer ID does not exist.");
+
+            CheeseType expectedCheeseType = order.getCheeseType();
+
+            Set<CheeseId> cheeseIds = order.getCheeses();
+            OrderId orderId = order.getOrderId();
+
```

Comment:
We can also check to ensure that an incomplete order's cheeseIds is empty.

If not, the app can start without errors or warnings even when there's an order in the data file like this:
![Screen Shot 2021-03-24 at 3 51 06 PM](https://user-images.githubusercontent.com/77544764/112273843-c8764f80-8cb8-11eb-8920-501970255ecb.png)